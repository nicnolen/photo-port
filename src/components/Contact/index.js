// Import dependencies
import React, { useState } from 'react';

// Make component function
function ContactForm() {
  const [formState, setFormState] = useState({
    name: '',
    email: '',
    message: '',
  });
  const { name, email, message } = formState;

  return (
    <section>
      <h1>Contact me</h1>
      <form id='contact-form'>
        <div>
          <label htmlFor='name'>Name:</label>
          <input type='text' name='name' defaultValue={name} />
        </div>
        <div>
          <label htmlFor='email'>Email address:</label>
          <input type='email' name='email' defaultValue={email} />
        </div>
        <div>
          <label htmlFor='message'>Message:</label>
          <textarea name='message' rows='5' defaultValue={message} />
        </div>
        <button type='submit'>Submit</button>
      </form>
    </section>
  );
}

// Export component
export default ContactForm;
