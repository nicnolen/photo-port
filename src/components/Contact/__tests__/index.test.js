// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('ContactForm component', () => {
  // verify the component is rendering properly
  it('Contact component renders', () => {
    render(<Contact />);
  });

  it('h1 tag renders', () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId('h1tag')).toHaveTextContent('Contact me'); // eslint-disable-line
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
