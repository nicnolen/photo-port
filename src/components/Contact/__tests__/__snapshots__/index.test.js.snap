// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Contact me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
